pyuvdata/utils.py

```diff
@@ -682,3 +682,99 @@ def _combine_histories(history1, history2):
+    """
+    Return redundant baseline groups, and lists of corresponding lengths and vectors
+
+    Args:
+        baseline_inds = Baseline indices.  (Nbls,), (int)
+        baseline_vecs = Baseline vectors in meters. (Nbls, 3), (float)
+        tol  = Absolute tolerance of redundancy, in meters. (float)
+
+    Returns:
+        baseline_groups: list of arrays of redundant baseline indices
+        vec_bin_centers: List of vectors describing redundant group centers
+        lengths: List of redundant group baseline lengths in meters
+
+    """
+    Nbls = baseline_inds.shape[0]
+
+    if not baseline_vecs.shape == (Nbls, 3):
+        raise ValueError("Baseline vectors must be shape (Nbls, 3)")
+
+    # For each baseline, list all others that are within the tolerance distance.
+
+    adj = {}   # Adjacency list
+
+    for bi, bv0 in enumerate(baseline_vecs):
+        key0 = baseline_inds[bi]
+        adj[key0] = []
+        for bj, bv1 in enumerate(baseline_vecs):
+            dist = np.linalg.norm(bv1 - bv0)
+            if dist < tol:
+                key1 = baseline_inds[bj]
+                adj[key0].append(key1)
+
+    # The adjacency list defines a set of graph edges.
+    # For each baseline b0, loop over its adjacency list ai \in adj[b0]
+    #   If adj[b0] is a subset of adj[ai], then ai is in a redundant group with b0
+
+    bl_gps = []
+    for k in adj.keys():
+        a0 = adj[k] + [k, ]
+        group = [k]
+        for a in a0:
+            if set(a0).issubset(adj[a]) and a not in group:
+                group.append(a)
+        group.sort()
+        bl_gps.append(group)
+
+    # We end up with multiple copies of each redundant group, so remove duplicates
+    bl_gps = np.unique(bl_gps)
+
+    N_unique = len(bl_gps)
+    vec_bin_centers = np.zeros((N_unique, 3))
+    for gi, gp in enumerate(bl_gps):
+        inds = [np.where(i == baseline_inds)[0] for i in gp]
+        vec_bin_centers[gi] = np.mean(baseline_vecs[inds, :], axis=0)
+
+    lens = np.sqrt(np.sum(vec_bin_centers**2, axis=1))
+
+    return bl_gps, vec_bin_centers, lens
+
+
+def get_antenna_redundancies(antenna_numbers, antenna_positions, tol=1.0, include_autos=False):
+    """
+    Construct baselines from antenna positions and get baseline redundancies.
+
+    Args:
+        antenna_numbers:  Antenna numbers.   (float), shape (Nants,)
+        antenna_positions:  Antenna position vectors in meters. Cartesian frame  (float), shape (Nants, 3)
+        tol = (float)   Redundancy tolerance in meters. (float)
+        include_autos =  Include autocorrelations (bool). Default is false
+
+    Returns:
+        baseline_groups: list of arrays of redundant baseline indices
+        vec_bin_centers: List of vectors describing redundant group centers
+        lengths: List of redundant group baseline lengths in meters
+    """
+    Nants = antenna_numbers.shape[0]
+
+    bl_inds = []
+    bl_vecs = []
+
+    def antnum_to_baseline(ant1, ant2):
```

Comment:
This is replicating existing code, we should avoid doing that. If the issue is that you want to put this functionality in utils rather than uvdata, make a function in utils that is called by the uvdata method.